provider/aws/aws.go

```diff
@@ -939,6 +982,40 @@ func (p *AWSProvider) newChange(action route53types.ChangeAction, ep *endpoint.E
 	return change
 }
 
+func buildChangeForGeoProximityEndpoint(change *Route53Change, ep *endpoint.Endpoint) {
+	geoproximity := &route53types.GeoProximityLocation{}
+	useGeoproximity := false
+	if prop, ok := ep.GetProviderSpecificProperty(providerSpecificGeoProximityLocationAWSRegion); ok {
+		geoproximity.AWSRegion = aws.String(prop)
+		useGeoproximity = true
+	}
+	if prop, ok := ep.GetProviderSpecificProperty(providerSpecificGeoProximityLocationBias); ok {
+		bias, _ := strconv.ParseInt(prop, 10, 32)
+		geoproximity.Bias = aws.Int32(int32(bias))
+	}
+	if prop, ok := ep.GetProviderSpecificProperty(providerSpecificGeoProximityLocationCoordinates); ok {
+		coordinates := strings.Split(prop, ",")
+		if len(coordinates) == 2 {
+			latitude := coordinates[0]
+			longitude := coordinates[1]
+			geoproximity.Coordinates = &route53types.Coordinates{
+				Latitude:  aws.String(latitude),
+				Longitude: aws.String(longitude),
+			}
+			useGeoproximity = true
+		} else {
+			log.Errorf("Invalid coordinates format for %s: %s; expected format 'latitude,longitude'", providerSpecificGeoProximityLocationCoordinates, prop)
```

Comment:
This is debatable, but most likely should be log.Warn, to warn user so they need to change configuraiton

Reply:
I feel an `Errorf` here is okay, because the user is going to expect e-dns to create those records but it won't. That object with the invalid coordinates is just going to fail to reconcile.
We are also using an `Errorf` [here](https://github.com/prasadkatti/external-dns/blob/d587756d062c74af544aaa44696ad08d2b9275b5/provider/aws/aws.go#L896) even when the record will get created.